staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go

```diff
@@ -94,6 +96,14 @@ func createHandler(r rest.NamedCreater, scope RequestScope, typer runtime.Object
 		audit.LogRequestObject(ae, obj, scope.Resource, scope.Subresource, scope.Serializer)
 
 		userInfo, _ := request.UserFrom(ctx)
+		// On create, get name from new object if unset
+		if len(name) == 0 {
+			if metadata, err := meta.Accessor(obj); err == nil {
+				name = metadata.GetName()
```

Comment:
When are names generated.  Seems like we want to generated them *before* admission is checked.

Reply:
Right. This did populate object name when `CREATE` before `admission.Admit` is called.

Reply:
> Right. This did populate object name when CREATE before admission.Admit is called.

You need some tests.  A few things look broken to me:
 1. if I included the name on create (block above), then choose a different name in the object, this will run admission on the name I claim, not the name I used to get past authorization.
 2. Generate name isn't done until the storage function is called (`r.Create` I think). That means this does not actually give the name being used.

Tests that would catch both conditions are needed.

Reply:
I think both items from this comment are still outstanding.

Reply:
@deads2k Did not find a good way to test this. Any suggestions?

Reply:
@liggitt @deads2k Sorry for taking such a long time. Updated. PTAL. Thanks.